test/scripts/test_config.py

```diff
@@ -0,0 +1,323 @@
+
+        # Set configs, commit configs and then read configs
+        self.test_config_set(ser)
+        self.__send_config_message(
+            ser, message, wait_msg, pattern, False, None, self.COMMIT_SLEEP_TIME_S
+        )
+
+        self.test_config_get(ser)
+
+    def test_config_status(self, ser: SerialHelper):
+        """Send status message to all nodes
+
+        This tests the status message on all other nodes besides the
+        root node on the bus. Values are set from the TYPE_VALUE
+        dictionary and then the status message is sent for each node.
+        The response received is evaluated and the output of all
+        configuration keys is evaluated to the key of the configuration
+        key-pair values set. The response message is printed to the
+        console as:
+            Response msg -- Node Id: {node_id}
+            Num Keys: {num_keys}
+            key_1
+            key_2
+            key_3
+            ...
+
+        Args:
+            ser (SerialHelper): Serial helper instance passed in from
+                                starting the pytests (see: conftest.py).
+        """
+        pattern = r"Response msg -- Node Id: [0-9a-fA-F]{16}, Partition: \d+, Commit Status: \d+\nNum Keys: (\d+)\n"
+        wait_msg = "Successful status request send\n"
+
+        def message(x, y, z, a):
+            return "bm cfg status " + x + " s\n"
+
+        def compare(ser: SerialHelper, read: str, value: Any):
+            """Compares status message keys to known set keys
+
+            Checks all keys on the node and determines if the keys
+            exist on the
+            """
+            count = int(re.search(pattern, read).group(1))
+            found = 0
+            for i in range(count):
+                read_key = ser.read_line()
+                for value in self.TYPE_VALUE.values():
+                    key = value[1]
+                    if key in read_key:
+                        found += 1
+            assert len(self.TYPE_VALUE) == found
+
+        self.test_config_set(ser)
+        self.__send_config_message(ser, message, wait_msg, pattern, False, compare)
+
+    def test_config_delete(self, ser: SerialHelper):
+        """Send delete message to all nodes
+
+        This test deletes all the keys in the TYPE_VALUE dictionary and
+        ensures that the message is sent properly.
+
+        Args:
+            ser (SerialHelper): Serial helper instance passed in from
+                                starting the pytests (see: conftest.py).
+        """
+        pattern = None
+        wait_msg = "Successfully sent del key request\n"
+
+        def message(x, y, z, a):
+            return "bm cfg del " + x + " s " + a + "\n"
+
+        # Currently there is no way to check if a node deleted a key
+        # other than calling bm cfg get <node_id> <partition> <key>
+        # and ensuring there is no response from the node in
+        # question, which is a manual process
```

Comment:
Couldn't we send cfg status and check whether the key is in the response?

Reply:
Ahhh very good callout! I didn't even think of that.

Reply:
Addressed and added this check!